### DISCUSSION TITLE
Question/problem regarding use of Ultrasonic sensor for EV3 MicroPython program

### DISCUSSION THREAD

**Discussion by jhess195** [pybricks/support#991] [Q&A]
Hello.  I'm using the EV3 MicroPython extension on VSC, and testing the use of the ultrasonic sensor via an obstacle detection and avoidance program.

Specifically, I'm trying to run the following example program:

https://pybricks.com/ev3-micropython/examples/robot_educator_ultrasonic.html

I have followed all the steps to get EV3 MicroPython on micro SD card, etc.  (https://pybricks.com/install/mindstorms-ev3/installation/). All of that is working, and I can create a program, download to EV3 and run, for basic motor movements, etc.

However, I run into a problem when trying to use the Ultrasonic sensor.

Python is returning an error for this line of the program:

**while obstacle_sensor.distance() > 300:**

The error message is included in the uploaded file, as well as repeated below:

**TypeError: '>' not supported between instances of 'NoneType' and 'int'**

I previously initialized the sensor:  **obstacle_sensor = UltrasonicSensor(Port.S4)**

and did the import: **from pybricks.ev3devices import Motor, UltrasonicSensor**

etc.

The error I'm getting back in Python is basically saying the sensor is not reading a value and you cannot compare no value to 300.

That was confusing to me because I didn't understand why the sensor would not be getting and providing a value.

If I go to the EV3 and scroll through to get to the section where I can read the Ultrasonic sensor values on port 4, it reads values just fine, in centimeters.

I would also note that the program returned an error message in VSC, as well as did not run on the EV3 after being downloaded.

**So my question is** ... is there some additional code I need to write to get this to work, to get the sensor to read a value and provide the value for use in the program?  Import something else into the program from pybricks, etc?

This seems like a very simple program that should work, but I feel like I'm missing something here.

Interestingly, I also have the Lego Robot Inventor set, and I used the code.pybricks.com site to write a simple object detection program and it worked perfect.

If anybody could please share some insight on how to get this object detection and avoidance program working on an EV3 using VSC and the EV3 MicroPython extension, I would most grateful, thank you!! : )

![Screenshot 2023-03-16 211426](https://user-images.githubusercontent.com/95370358/225795383-1be8a9af-5e7e-4fa2-af6a-cf42e3f74958.jpg)

**Comment by dlech** [comment]
It looks like you are running the program on your computer instead of on the EV3. Use the debug feature in VS Code to run the program on your EV3.

**Reply by jhess195** [reply]
![Screenshot 2023-03-16 220123](https://user-images.githubusercontent.com/95370358/225801616-8d6455d6-aed9-44bf-9dd5-ee9f0c8c177b.jpg)

Thanks for responding.  I think I did that, I connected the EV3 via the EV3 Device Browser and then made sure I had the latest copy of the obstacleAvoidance program on the EV3 and then ran that program from the EV3 and got the resulting error messages (i.e. Exited with error code).

**Reply by dlech** [reply]
For some reason, it isn't fixing the line endings for you. In the bottom blue bar in VS Code, click on `CRLF` and change it to `LF`. This will need to be done for every Python file.

**Reply by jhess195** [reply]
I cleaned up my file organization in VSC & EV3, then made the change you recommended, saved, re-uploaded and tried again.  The program worked perfect!

I had no idea what that LF vs CRLF "Select End of Line Sequence" parameter even was in VSC, so thank you very much for catching that and recommending the change! : )

![Screenshot 2023-03-16 222948](https://user-images.githubusercontent.com/95370358/225806026-d4f3adf0-ad0b-403c-82e5-ee5f1e7f7688.jpg)
